### ISSUE TITLE
handle application errors

### ISSUE THREAD

**Issue by grosser** [stve/airbrake-api#16]
When airbrake returns an application error it should either `puts "Application error"` or raise and tell me what happened.
Alternatively and maybe even simpler, dump the xml when a weird error like this appears, its ugly but always accurate :)

atm it just fails with a mysterious:

```
/Users/mgrosser/.rvm/gems/ruby-1.9.3-p194/gems/multi_xml-0.5.1/lib/multi_xml/parsers/nokogiri.rb:15:in `parse': Opening and ending tag mismatch: meta line 4 and head (MultiXml::ParseError)
    from /Users/mgrosser/.rvm/gems/ruby-1.9.3-p194/gems/multi_xml-0.5.1/lib/multi_xml.rb:119:in `parse'
    from /Users/mgrosser/.rvm/gems/ruby-1.9.3-p194/gems/airbrake-api-3.2.2/lib/airbrake-api/client.rb:39:in `xml'
    from /Users/mgrosser/.rvm/gems/ruby-1.9.3-p194/gems/httparty-0.8.3/lib/httparty/parser.rb:140:in `parse_supported_format'
    from /Users/mgrosser/.rvm/gems/ruby-1.9.3-p194/gems/httparty-0.8.3/lib/httparty/parser.rb:102:in `parse'
    from /Users/mgrosser/.rvm/gems/ruby-1.9.3-p194/gems/httparty-0.8.3/lib/httparty/parser.rb:66:in `call'
    from /Users/mgrosser/.rvm/gems/ruby-1.9.3-p194/gems/httparty-0.8.3/lib/httparty/request.rb:241:in `parse_response'
    from /Users/mgrosser/.rvm/gems/ruby-1.9.3-p194/gems/httparty-0.8.3/lib/httparty/request.rb:211:in `block in handle_response'
    from /Users/mgrosser/.rvm/gems/ruby-1.9.3-p194/gems/httparty-0.8.3/lib/httparty/response.rb:18:in `call'
    from /Users/mgrosser/.rvm/gems/ruby-1.9.3-p194/gems/httparty-0.8.3/lib/httparty/response.rb:18:in `parsed_response'
    from /Users/mgrosser/.rvm/gems/ruby-1.9.3-p194/gems/httparty-0.8.3/lib/httparty/response.rb:51:in `method_missing'
    from /Users/mgrosser/.rvm/gems/ruby-1.9.3-p194/gems/hashie-1.2.0/lib/hashie/mash.rb:127:in `deep_update'
    from /Users/mgrosser/.rvm/gems/ruby-1.9.3-p194/gems/hashie-1.2.0/lib/hashie/mash.rb:54:in `initialize'
    from /Users/mgrosser/.rvm/gems/ruby-1.9.3-p194/gems/airbrake-api-3.2.2/lib/airbrake-api/client.rb:26:in `new'
    from /Users/mgrosser/.rvm/gems/ruby-1.9.3-p194/gems/airbrake-api-3.2.2/lib/airbrake-api/client.rb:26:in `fetch'
    from /Users/mgrosser/.rvm/gems/ruby-1.9.3-p194/gems/airbrake-api-3.2.2/lib/airbrake-api/notice.rb:11:in `find'
    from /Users/mgrosser/.rvm/gems/ruby-1.9.3-p194/gems/airbrake-api-3.2.2/lib/airbrake-api/notice.rb:35:in `block in find_all_by_error_id'
    from /Users/mgrosser/.rvm/gems/ruby-1.9.3-p194/gems/parallel-0.5.16/lib/parallel.rb:275:in `call'
    from /Users/mgrosser/.rvm/gems/ruby-1.9.3-p194/gems/parallel-0.5.16/lib/parallel.rb:275:in `call_with_index'
    from /Users/mgrosser/.rvm/gems/ruby-1.9.3-p194/gems/parallel-0.5.16/lib/parallel.rb:130:in `block (2 levels) in work_in_threads'
    from /Users/mgrosser/.rvm/gems/ruby-1.9.3-p194/gems/parallel-0.5.16/lib/parallel.rb:123:in `loop'
    from /Users/mgrosser/.rvm/gems/ruby-1.9.3-p194/gems/parallel-0.5.16/lib/parallel.rb:123:in `block in work_in_threads'
    from /Users/mgrosser/.rvm/gems/ruby-1.9.3-p194/gems/parallel-0.5.16/lib/parallel.rb:14:in `block (2 levels) in in_threads'
```

Html is:

``` Html
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
    <html>
    <head>
      <meta http-equiv="content-type" content="text/html; charset=ISO-8859-1">
      <style type="text/css">
        html, body, iframe { margin: 0; padding: 0; height: 100%; }
        iframe { display: block; width: 100%; border: none; }
      </style>
    <title>Application Error</title></head>
    </head>
    <body>
      <iframe src="https://s3.amazonaws.com/heroku_pages/error.html">
        <p>Application Error</p>
      </iframe>
    </body>
    </html>
```

**Comment by stve** [comment]
Yuck, that is ugly.  Any guesses as to what the response code might be?

The refactor branch has better response code handling so I'll likely just raise when non-200's are received.  Probably going to release in the next day or so.

**Comment by grosser** [comment]
didnt check the code but my guess is its 500 :)

if you raise, raise with e.g. the content of the body, so I dont have to go in and edit the gem all the time :D

**Comment by stve** [comment]
I just released 4.0.0 which has better http code error handling.  Docs need a little work, but it's basically a re-write using Faraday and FaradayMiddleware.
